website/blog/2023-09-05-wrapper-as-a-standalone-component.md

```diff
@@ -0,0 +1,157 @@
+---
+title: How to use the Wrapper as a standalone component
+description: Explains the process of using the Wrapper component in WordPress templates.
+slug: wrapper-as-a-standalone-component
+authors: kancijan
+
+date: 2023-09-05
+tags: [eightshift, boilerplate, wrapper, components]
+hide_table_of_contents: false
+---
+
+As one of the most powerful features in the Eightshift DevKit, the Wrapper is a component used for every block in the Gutenberg editor, but what about WordPress templates?
+<!--truncate-->
+
+:::note
+[The Wrapper](https://eightshift.com/docs/basics/blocks-wrapper/) is designed to be the ultimate top-level component that controls how your block behaves in the website layout. The Wrapper is a sort of a 'section' in traditional builders. By default, you can control a whole lot of stuff, but there is an option to add your custom attributes and fine-tune the Wrapper to your project's needs.
+:::
+
+## What are WordPress templates?
+
+Before the days of [Full Site Editing](https://developer.wordpress.org/block-editor/getting-started/full-site-editing/) in WordPress, we used [template files](https://developer.wordpress.org/themes/basics/template-hierarchy/) to add custom layout to a blog page or post archives.
+
+Depending on your setup, you might still use those as they are still a core part of WordPress themes.
+
+## Usage in templates
+
+Thanks to the `Components` helper we can easily `render()` any component in our template. 
+
+:::Note
+Interested in how to use components in a block? Take a look at [our docs](https://eightshift.com/docs/basics/blocks-component-in-block#i-have-a-component-that-i-want-to-use-manually).
+:::
+
+For this example, we'll use `index.php` as you probably already have it in your theme. If not, use the code below.
+
+```php
+<?php
+
+/**
+ * Display regular index/home page
+ *
+ * @package InfinumLibs
+ */
+
+use InfinumLibsVendor\EightshiftLibs\Helpers\Components;
+
+get_header();
+
+if (have_posts()) {
+	while (have_posts()) {
+		the_post();
+		the_title();
+		the_content();
+	}
+}
+
+get_footer();
+```
+![No Wrapper](/img/blog/no-wrapper.png)
+
+And here it is, a bone stock blog page that you can style only `the old way`... But not in our case!
+
+## Using the Wrapper component
+
+First, let's import our `Components` helper class. Make sure to use the namespace from your `composer.json`.
+```php
+use InfinumLibsVendor\EightshiftLibs\Helpers\Components;
+```
+
+If we take a deeper look into the `render` method, we'll see that for the `$component` parameter we can pass `component's name or the full path (ending with .php)`, and since the Wrapper is outside of the `components` or `custom` folder, we have to use `wrapper.php` combined with `Components::getProjectPaths('blocksDestinationWrapper')` for the `$parentPath` parameter.
```

Comment:
```suggestion
If we take a deeper look into the `render` method, we'll see that for the `$component` parameter we can pass the _component's name or the full path (ending with .php)_, and since the Wrapper is outside of the `components` or `custom` folder, we have to use `wrapper.php` combined with `Components::getProjectPaths('blocksDestinationWrapper')` for the `$parentPath` parameter.
```